packages/host/app/services/matrix-service.ts

```diff
@@ -641,23 +638,33 @@ export default class MatrixService extends Service {
     });
   }
 
-  async addCardsToRoom(cards: CardDef[], roomId: string) {
-    let cardEventIds = await this.cardEventPublisher.addCardsToRoom(
-      cards,
-      roomId,
-    );
-    return cardEventIds;
+  async downloadCardFileDef(cardFileDef: FileAPI.SerializedFile) {
+    return this.fileDefManager.downloadCardFileDef(cardFileDef);
   }
 
-  async addSkillCardsToRoomHistory(
-    skillCards: SkillCardModule.SkillCard[],
-    roomId: string,
+  async downloadFile(url: string) {
+    let response = await this.network.fetch(url, {
+      headers: {
+        Authorization: `Bearer ${this.client.getAccessToken()}`,
+      },
+    });
+    if (!response.ok) {
+      throw new Error(`HTTP error. Status: ${response.status}`);
+    }
+    return response;
+  }
+
+  async uploadCards(cards: CardDef[]) {
+    let cardFileDefs = await this.fileDefManager.uploadCards(cards);
+    return cardFileDefs;
+  }
+
+  async uploadCommandDefinitions(
+    commandDefinitions: SkillCardModule.CommandField[],
   ) {
-    let cardEventIds = await this.cardEventPublisher.addSkillCardsToRoomHistory(
-      skillCards,
-      roomId,
-    );
-    return cardEventIds;
+    let commandFileDefs =
+      await this.fileDefManager.uploadCommandDefinitions(commandDefinitions);
```

Comment:
Perhaps fileDefManager should be part of our "ExtendedClient" matrix interface to allow for easier mocking in host?

Reply:
I moved `downloadContent` to `extendedClient` to make it easier to mock file downloads. I also renamed `fileDefManager` to `cardUploader`, and kept the upload logic there since it depends on several services that are not available in extendedClient.